hoomd/md/pytest/test_gsd.py

```diff
@@ -0,0 +1,369 @@
+import hoomd
+import numpy as np
+import pytest
+try:
+    import gsd.hoomd
+except ImportError:
+    pytest.skip("gsd not available", allow_module_level=True)
+
+from hoomd.pytest.test_snapshot import assert_equivalent_snapshots
+
+
+@pytest.fixture(scope='function')
+def hoomd_snapshot(lattice_snapshot_factory):
+    snap = lattice_snapshot_factory(particle_types=['t1', 't2'],
+                                    n=10, a=2.0)
+    if snap.exists:
+        typeid_list = [0] * int(snap.particles.N / 2)
+        typeid_list.extend([1] * int(snap.particles.N / 2))
+        snap.particles.typeid[:] = typeid_list[:]
+        snap.particles.velocity[:] = np.tile(np.linspace(1, 2, 3),
+                                            (snap.particles.N, 1))
+        snap.particles.acceleration[:] = np.tile(np.linspace(1, 2, 3),
+                                                (snap.particles.N, 1))
+        snap.particles.mass[:] = snap.particles.N * [1]
+        snap.particles.charge[:] = snap.particles.N * [2]
+        snap.particles.diameter[:] = snap.particles.N * [0.5]
+        snap.particles.angmom[:] = snap.particles.N * [[0, 0, 0, 1]]
+
+        # bonds
+        snap.bonds.types = ['b1', 'b2']
+        snap.bonds.N = 2
+        snap.bonds.typeid[:] = [0, 1]
+        snap.bonds.group[0] = [0, 1]
+        snap.bonds.group[1] = [2, 3]
+
+        # angles
+        snap.angles.types = ['a1', 'a2']
+        snap.angles.N = 2
+        snap.angles.typeid[:] = [1, 0]
+        snap.angles.group[0] = [0, 1, 2]
+        snap.angles.group[1] = [2, 3, 0]
+
+        # dihedrals
+        snap.dihedrals.types = ['d1']
+        snap.dihedrals.N = 1
+        snap.dihedrals.typeid[:] = [0]
+        snap.dihedrals.group[0] = [0, 1, 2, 3]
+
+        # impropers
+        snap.impropers.types = ['i1']
+        snap.impropers.N = 1
+        snap.impropers.typeid[:] = [0]
+        snap.impropers.group[0] = [3, 2, 1, 0]
+
+        # constraints
+        snap.constraints.N = 1
+        snap.constraints.group[0] = [0, 1]
+        snap.constraints.value[0] = 2.5
+
+        # special pairs
+        snap.pairs.types = ['p1', 'p2']
+        snap.pairs.N = 2
+        snap.pairs.typeid[:] = [0, 1]
+        snap.pairs.group[0] = [0, 1]
+        snap.pairs.group[1] = [2, 3]
+
+    return snap
+
+
+def lj_integrator():
+    integrator = hoomd.md.Integrator(dt=0.005)
+    lj = hoomd.md.pair.LJ(nlist=hoomd.md.nlist.Cell(), r_cut=2.5)
+    lj.params.default = {'sigma': 1, 'epsilon': 1}
+    integrator.forces.append(lj)
+    langevin = hoomd.md.methods.Langevin(hoomd.filter.All(), kT=1)
+    integrator.methods.append(langevin)
+
+    return integrator
+
+
+@pytest.fixture(scope='function')
+def create_md_sim(simulation_factory, device, hoomd_snapshot):
```

Comment:
I would just name this `simulation` and not bother with the `sim = create_md_sim` lines in the tests.